runtime/major_gc.c

```diff
@@ -285,7 +285,7 @@ Caml_inline void prefetch_block(value v)
      somewhere between 1/8-1/2 of a prefetch operation (in expectation,
      depending on alignment, word size, and cache line size), which is
      cheap enough to make this worthwhile. */
-  caml_prefetch(Hp_val(v));
+  caml_prefetch((void*)Hp_val(v));
   caml_prefetch((void*)&Field(v, 3));
```

Comment:
Technically, you could convert to `const void *` here, since that's what `__builtin_prefetch` expects.  The `const` is somewhat informative ("`caml_prefetch` doesn't change the value at this address") but probably meaningless in terms of memory model and concurrent operations.  Do as you prefer.

Reply:
I am very much in favor of more informative types. Thank you, done in 1d7f71f.